### ISSUE TITLE
local-up-cluster.sh always fail with "Timed out waiting for apiserver"

### ISSUE THREAD

**Issue by mtanino** [kubernetes/kubernetes#49319]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug
#49072

**What happened**:

After this commit was introduced, local-up-cluster.sh always fail with "Timed out waiting for apiserver" error.

```
commit 413ab26df92c3da66bc6eb60c1d1105f6ac267fc
    use https to check healthz in hack/local-up-cluster.sh
```

I'm not sure the reason but the local-up-cluster.sh successfully start if I specify ENABLE_RBAC=true option to local-up-cluster.sh.

**What you expected to happen**:

The local-up-cluster.sh will successfully start without any additional options.

**How to reproduce it (as minimally and precisely as possible)**:
```
% hack/local-up-cluster.sh
```

**Comment by RenaudWasTaken** [comment]
I've noticed that If you revert the change, once the cluster is started curling the healthz in https works

```
$ curl https://127.0.0.1:6443/healthz:6443/
healthz
curl: (60) server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
```

It might be a waiting problem

**Comment by mtanino** [comment]
@RenaudWasTaken

The patch applied following two fixes. Seems extending wait time fixes my original problem on #47739, but I'm not sure why API endpoint was changed to https.

- Wait 20 sec for starting API server
- Change API server's endpoint from http to https.
